### PR TITLE
Add no-conflict function

### PR DESCRIPTION
**Pull request by lpil**
This allows the user to use this function without poluting the global namespace. :)

### CODE REVIEW

File: dist/pegasus-amd.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ function pegasus(a, xhr) {
 
   // Return request
   return xhr;
-}
+};
+
 // AMD support
 define('pegasus', [], function() { return pegasus; });
```
Comment (malukenho) [inline dist/pegasus-amd.js]: can you fix this EOF EOL too?

File: dist/pegasus-commonjs.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ function pegasus(a, xhr) {
 
   // Return request
   return xhr;
-}
+};
+
 // CommonJS
 module.export = pegasus;
```
Comment (malukenho) [inline dist/pegasus-commonjs.js]: Same above

**Comment by lpil** [comment]
Thanks :)

That's vim being opinionated there I think! How's that now?

**Comment by typicode** [comment]
Hi thanks for the PR :) and sorry for the delay.
I may be wrong or missing something but I'm not sure it's needed since pegasus is the first lib that should be loaded.

Standard scenario

```html
<script src="pegasus.min.js"></script>
<script>
  // no risk of conflict here
  var request = pegasus('http://example.com')
</script>
<script src="other.js"></script>
<!-- if other.js override pegasus, it shouldn't be a problem
     as pegasus is usually not needed after -->
```

In case you're using pegasus after `other.js` and it's been overridden by it

```html
<script src="pegasus.min.js"></script>
<script>
  var _pegasus = pegasus
  var request = _pegasus('http://example.com')
</script>
<script src="other.js"></script>
```

However, you're right about the current encapsulation in AMD and CommonJS versions, it's not well done and polluting the global namespace. I'll fix that.

**Comment by typicode** [comment]
I've updated [pegasus-amd.js](https://raw.githubusercontent.com/typicode/pegasus/v0.3.0/dist/pegasus-amd.js) to encapsulate pegasus function so that it doesn't pollute the global namespace as it was before.

**Comment by lpil** [comment]
This doesn't help people not using amd.

**Comment by typicode** [comment]
What do you mean?
I'm not sure to really understand the issue.

**Comment by lpil** [comment]
This pull request added a no-conflict option for those not using AMD.

**Comment by typicode** [comment]
Ok, I think I'm getting it. The use case would be:

```html
<script src="other-lib.js"></script>
<!-- 
other-lib.js defines a function called pegasus:
function pegasus () { ... })
-->
<script src="pegasus-noclonfict.js"></script>
<script>
  var request = pegasusNoConflict()('/data.json')
</script>
...
<script>
  pegasus() // will use pegasus from other-lib.js
</script>
```

**Comment by lpil** [comment]
Yup! Sorry I didn't make it clear.

**Pull request closed without merging**